analyzers/tests/SonarAnalyzer.UnitTest/TestCases/VariableShadowsField.CSharp12.cs

```diff
@@ -1,11 +1,33 @@
-﻿class MyClass(int first, int second, int third, int fourth)
-{
-    private int first = first; // Compliant
-    private int second; // FN
+﻿using System;
 
-    void MyMethod()
+class MyParentClass(int parentClassParam)
+{
+    class MyClass(int a, int b, int c, int d, int e, int f, int g, int h)
     {
-        int third = 1; // FN
-        _ = first is object fourth; // FN
+        public int a = a; // Compliant
+        public int b; // FN
+
+        public int c { get; set; } = c; // Compliant
+        public int d { get; set; } // FN
+
+        public int g = a; // FN {{Rename 'g' which hides the primary constructor parameter with the same name.}}
+
+        class MyChildClass(int childClassParam) { }
+
+        MyClass(int ctorParam, int f) : this(ctorParam, f, 1, 1, 1, 1, 1, 1) { } // Compliant
+
+        void MyMethod()
+        {
+            int parentClassParam = 42; // Compliant
```

Comment:
Is this a FN?

Reply:
Nope, it's error CS9105 if you try to access the primary ctor param in this context ([example](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgAQCYBGAWACgKiACAWQE8AFCAJwFMYEBhAG1RQAUsBNQAOrDtz4oUzFhDABKCgG8K1DdRoNe/IZ2oRl5TdTUnTm/ABY69WmwQALAPYYBxy5vNevAN1ZDagBeMQlOXRk5BQBudV8AX3iNJPJUoA==))